### PR TITLE
Add node/index, move node/user guide to sep file

### LINKED ISSUE #54: Move perun node user guide to a separate page.
Currently, all the sections of perun-node documentation are contained in a single file `node/introduction.rst`.

It is difficult to extend this while adding other sections such as developer documentation.

So, move the user guide to a separate file, while ensuring it appears as a sub section of the perun-node in the Table of Contents in the landing page.

### PR DESCRIPTION
**Pull request by manoranjith**
- Adding an index page for node documentation allows different sections
  to be split into individual files.

- Move the user guide section into a separate file.

Closes #54

### CODE REVIEW

File: source/node/user_guide.rst [hunk 1]
```diff
@@ -0,0 +1,552 @@
...
+.. code-block::
+
+  > # [In Alice's CLI]
+  > channel close-n-settle-on-chain ch_1_bob
+
+.. note::
+
+  **Collaborative and Non collaborative channel close:**
+
+  When any one of the channel participant sends a channel close request to
```
Comment (anagha-ks) [inline source/node/user_guide.rst:441]:
```suggestion
  When any one of the channel participants sends a channel close request to
```

**Review by anagha-ks — CHANGES_REQUESTED** [review]

File: source/node/user_guide.rst [hunk 2]
```diff
@@ -0,0 +1,552 @@
...
+
+To start a perun-node, user needs an ID provider, wallet provider, blockchain
+node and a configuration file. These artifacts and the order in which they have
+to be set up is shown in the below diagram.
+
+.. image:: ../_generated/node/act_node_init.svg
+  :align: Center
+  :alt: Image not available
+
+1. The blockchain node was setup in step 1 and the perun-node binary was
```
[the remark this answers is not in the document]
Comment (manoranjith) [reply source/node/user_guide.rst]: Have revised this line in the latest commit.

File: source/node/user_guide.rst [hunk 3]
```diff
@@ -0,0 +1,553 @@
...
+  - **>**, execute them in perun-node cli.
+
+.. _Getting Started Perun Node:
+
+Getting started
+---------------
+
+1. Start a blockchain network using ganache-cli node using the below command.
+   The two accounts in the command correspond to accounts that will be used in
+   default configuration artifacts which we will be generate in later steps.
```
Comment (anagha-ks) [inline source/node/user_guide.rst:55]:
```suggestion
   default configuration artifacts which we will generate in later steps.
```

File: source/node/user_guide.rst [hunk 4]
```diff
@@ -0,0 +1,553 @@
...
+   We fund each of these accounts are with 10 ETH each.
```
Comment (anagha-ks) [inline source/node/user_guide.rst:56]:
```suggestion
   We fund each of these accounts with 10 ETH each.
```

**Review by anagha-ks — CHANGES_REQUESTED** [review]

**Review by anagha-ks — APPROVED** [review]

**Pull request merged**